src/Paramore.Brighter.Outbox.MsSql/MsSqlOutbox.cs

```diff
@@ -499,7 +499,7 @@ public void MarkDispatched(Guid id, DateTime? dispatchedAt = null, Dictionary<st
         
         private void CreatePagedDispatchedCommand(SqlCommand command, double millisecondsDispatchedSince, int pageSize, int pageNumber)
         {
-            var pagingSqlFormat = "SELECT * FROM (SELECT ROW_NUMBER() OVER(ORDER BY Timestamp DESC) AS NUMBER, * FROM {0}) AS TBL WHERE DISPATCHED IS NOT NULL AND DISPATCHED < DATEADD(millisecond, @OutStandingSince, getdate()) AND NUMBER BETWEEN ((@PageNumber-1)*@PageSize+1) AND (@PageNumber*@PageSize) ORDER BY Timestamp DESC";
+            var pagingSqlFormat = "SELECT * FROM (SELECT ROW_NUMBER() OVER(ORDER BY Timestamp DESC) AS NUMBER, * FROM {0}) AS TBL WHERE DISPATCHED IS NOT NULL AND DISPATCHED < DATEADD(millisecond, @OutStandingSince, getutcdate()) AND NUMBER BETWEEN ((@PageNumber-1)*@PageSize+1) AND (@PageNumber*@PageSize) ORDER BY Timestamp DESC";
```

Comment:
If I look at the docs for MySQL 8.0 it says to use UTC_DATE() (see https://dev.mysql.com/doc/refman/8.0/en/date-and-time-functions.html#function_utc-date). It is interesting because getdate() is not listed but curdate() as the alternative, which leads to the 'why did this ever work' question, which could be because it supports the SQL syntax??

So should we be using utc_date??

Reply:
Thanks for sharing the docs, we should not be using **utc_date()** because it doesn't have enough precision.

MySQL workbench doesn't even recognize this functions, I have no idea why it works in our integration tests.
FYI, the same issue occurs when we are using DATEADD, whereas the official functions is DATE_ADD.

Just to be safe, I moved to the official functions: DATE_ADD and UTC_TIMESTAMP and pushed the changes.

I also noticed the implementation for MarkDispatched was wrong, as MSSQL and MySQL outbox weren't actually defaulting to UTC Now (it would just leave it NULL if that's what was passed as input).